static/js/message_list_data.js

```diff
@@ -112,74 +112,73 @@ MessageListData.prototype = {
 
         this._items = [];
         this._hash.clear();
-    },
+    }
 
     get(id) {
         id = parseFloat(id);
         if (isNaN(id)) {
             return;
         }
         return this._hash.get(id);
-    },
+    }
 
     clear_selected_id() {
         this._selected_id = -1;
-    },
+    }
 
     selected_id() {
         return this._selected_id;
-    },
+    }
 
     set_selected_id(id) {
         this._selected_id = id;
-    },
+    }
 
     selected_idx() {
         return this._lower_bound(this._selected_id);
-    },
+    }
 
     reset_select_to_closest() {
         this._selected_id = this.closest_id(this._selected_id);
-    },
+    }
 
     is_search() {
         return this.filter.is_search();
-    },
+    }
     can_mark_messages_read() {
         return this.filter.can_mark_messages_read();
-    },
+    }
     _get_predicate() {
         // We cache this.
         if (!this.predicate) {
             this.predicate = this.filter.predicate();
         }
         return this.predicate;
-    },
+    }
 
     valid_non_duplicated_messages(messages) {
         const predicate = this._get_predicate();
-        const self = this;
-        return messages.filter((msg) => self.get(msg.id) === undefined && predicate(msg));
-    },
+        return messages.filter((msg) => this.get(msg.id) === undefined && predicate(msg));
```

Comment:
Is this change correct?  In any case I think it's just a bad idea to use `this` in this sort of inner logic; might be cleanest to do `const get = this.get` like we do with `predicate` for consistency.

Reply:
`this.get()` is fine inside an arrow function. That’s why I converted all the inner functions to arrow functions. My view is that we should get comfortable with this style rather than shying away from it.

Meanwhile `get = this.get; get()` is not fine because it fails to bind `this`.

Reply:
I guess it seems likely this is correct; it may be the historical problems I'm recalling with use of `this` were caused precisely because these things weren't classes and various older loop iteration methods would rebind `this` inside the loops in a way that is no longer possible.  I just do remember a lot of time being spent with `this`-related scope issues in this module (long ago!).

Reply:
Ahh, OK, so it would be a problem if these weren't arrow functions?  I suppose I can live with that situation.

Reply:
Right.